### DISCUSSION TITLE
Tagesdaten um Mitternacht löschen

### DISCUSSION THREAD

**Discussion by uli-bs** [tbnobody/OpenDTU#1779] [Q&A] [answered]
Hier laufen 3 WR an openDTU oB, 2 davon mit Modulen, einer an einem Akku.
Der mit Akku hat morgens lt. openDTU immer schon fleißig gearbeitet, obwohl die Klemmspannung unter der Abschaltschwelle liegt.
Hat er natürlich nicht wirklich, es wird einfach der Vortagesertrag nicht gelöscht, weil ja auch über Nacht Spannung anlag (so lese ich auch die Hilfestellung im Einrichtungsmenu).

Ich meine, dass ich dieses Verhalten früher (k.A. welche Version das war) nicht beobachtet habe.
Mache ich jetzt etwas falsch, oder gibt es tatsächlich keine Option den Tagesertrag um 0:00 zu resetten, wenn der WR an "Dauerstrom" hängt? Dann würde ich um dieses Feature bitten.

**Comment by MetaChuh** [comment]
hi @uli-bs 

> Ich meine, dass ich dieses Verhalten früher (k.A. welche Version das war) nicht beobachtet habe.

doch, war immer schon so, da der inverter da der inverter die daten liefert.
auch mit opendtu yield day correction, reicht's wenn du die dtu rebootest (zb durch update, oder selbst per watchdog) und du hast wieder die inverterdaten drin, da sie neu eingelesen werden.

> Mache ich jetzt etwas falsch

nein, du machst nichts falsch.
du musst nur jeden tag um 23:59 einen reboot befehl via opendtu an deinen batterie inverter schicken, da das bisher die einzige bekannte möglichkeit ist, den yield day wert inverterseitig zurückzusetzen.

(info: 23:59 statt 00:00h für den reboot, weil sonst die reboot phase nicht vor 00:00h durchgeführt wurde, und darum auch nach dtu reset yield day at midnight, der vorherige inverter wert wieder eingelesen wird ... also bitte immer zuerst inverter reset, dann dtu reset)

wenn du code snipplets oder hilfe brauchst, wie du den inverter per api restartest, sag kurz bescheid, für welche programmiersprache, oder shell sprache

glg
metachuh

**Comment by MetaChuh** [comment]
ps info:
falls du den inverter reboot lieber ohne externe steuerungen, direkt in der dtu durchführen willst:

das opendtu addon mod projekt [`OpenDTU-OnBattery`](https://github.com/helgeerbe/OpenDTU-OnBattery/tree/master) beinhaltet die inverter reboot at midnight funktion bereits.

du kannst auch zwischen beiden opendtu varianten hin und her flashen, wenn du datumsgleiche oder maximal leicht datumsunterschiedliche releases verwendest.

alle opendtu daten bleiben auch beim hin und her flashen vorhanden.

**Reply by uli-bs** [reply]
@MetaChuh 
Vielen Dank für den Hinweis, wenn sich das innerhalb von openDTU oB lösen lässt, wäre mir natürlich am Liebsten. 
Über den von dir angegebenen Link lande ich allerdings bei der von mir bereits eingesetzten Version.
Über /Dynamic Power limiter Settings/Wechselrichter Neustart/ kann ich aber nur "ganze" Stunden einstellen und mit "0:00" habe ich den beschriebenen Effekt.

**Reply by MetaChuh** [reply]
@uli-bs
danke dir für die info bezüglich vollen stunden bei oB 👍
ich arbeite mit externer, eigener steuerung, d'rum habe ich auch nur gewusst, dass die inverter reboot funktion bei @helgeerbe  bereits eingebaut ist.

vielleicht fragst du ihn, mit den informationen die du jetzt hast, was er für dich mit wenigem aufwand verbessern kann, ohne andere sachen in seinem projekt zu beeinträchtigen.

thx & greetings,
metachuh

**Comment by uli-bs** [comment, marked as answer]
Ich glaube, dass ich das (hausgemachte) Problem gefunden habe...
Der WR Neustart war für 0:00Uhr geplant (warum ist die Einstellung eigentlich unter "Power-Limiter Settings" wo sie doch die WRs betrifft?)
Gleichzeitig hatte ich wohl irgendwann (darum ging es auch früher) auch für diesen WR die "Tagesertragskorrektur" aktiviert...
Das Ergebnis war dann eben, dass der Ertrag innerhalb der DTU fortgeschrieben wurde.

**Reply by MetaChuh** [reply]
> Ich glaube, dass ich das (hausgemachte) Problem gefunden habe...

super, perfekt 👍

> (warum ist die Einstellung eigentlich unter "Power-Limiter Settings" wo sie doch die WRs betrifft?)

ich glaube der grund, warum diese einstellung nicht bei `inverter` hinzugefügt wurde, ist die modularität.
dadurch bleibt die sektion `inverter` identisch zum opendtu master, und ein merge von neueren opendtu versionen nach oB wird erleichtert, da diese segmente nicht gepatcht werden müssen.

> Gleichzeitig hatte ich wohl irgendwann (darum ging es auch früher) auch für diesen WR die "Tagesertragskorrektur" aktiviert...
Das Ergebnis war dann eben, dass der Ertrag innerhalb der DTU fortgeschrieben wurde.

richtig, yield day correction muss in diesem fall deaktiviert sein.
ist es bei einen batterie invertern auch 👍
